astroid/interpreter/_import/spec.py

```diff
@@ -453,8 +453,13 @@ def find_spec(modpath: list[str], path: Sequence[str] | None = None) -> ModuleSp
             _path, modname, module_parts, processed, submodule_path or path
         )
         processed.append(modname)
-        if modpath and isinstance(finder, Finder):
-            submodule_path = finder.contribute_to_path(spec, processed)
+        if modpath:
+            if isinstance(finder, Finder):
+                submodule_path = finder.contribute_to_path(spec, processed)
+            # If modname is a package from an editable install, update submodule_path
+            # so that the next module in the path will be found inside of it using importlib.
+            elif finder.__name__ in {"_EditableFinder", "_EditableNamespaceFinder"}:
```

Comment:
I'm wondering if we should extract this logic into a class, but for now I think this is fine!

Reply:
I agree. Do you think the logic will apply to all of the keys in _MetaPathFinderModuleTypes? (I'm not familiar with six). If so, an improvement for now would be to use that.

Reply:
When I implemented this @jacobtylerwalls actually (rightly) stressed that we should probably be careful here: we rather want to have less support for installers we don't know than making assumptions that don't turn out to be true.

We can always add support for new ones if we encounter them.

The import system is apparently not that well defined..

Reply:
Sounds good. I left the set alone, but moved it to be a top-level variable instead of treating it as a constant. Any new classes can be added to that set.